js_modules/dagit/packages/ui/src/components/ProductTour.tsx

```diff
@@ -0,0 +1,128 @@
+// eslint-disable-next-line no-restricted-imports
+import {Placement} from '@blueprintjs/popover2';
+import React from 'react';
+import styled, {CSSProperties} from 'styled-components/macro';
+
+import {Box} from './Box';
+import {Button} from './Button';
+import {Colors} from './Colors';
+import {Popover} from './Popover';
+import {Subheading} from './Text';
+
+export enum Position {
+  TOP_LEFT = 'top-start',
+  TOP_CENTER = 'top',
+  TOP_RIGHT = 'top-end',
+  BOTTOM_LEFT = 'bottom-start',
+  BOTTOM_CENTER = 'bottom',
+  BOTTOM_RIGHT = 'bottom-end',
+}
+
+type ObjectType =
+  | {img: string; video?: undefined; object?: undefined}
+  | {video: string; img?: undefined; object?: undefined}
+  | {object: React.ReactNode; video?: undefined; img?: undefined}
+  | {img?: undefined; video?: undefined; object?: undefined};
+
+type Props = {
+  title: React.ReactNode;
+  description?: React.ReactNode;
+  actions?: {
+    next?: () => void;
+    dismiss?: () => void;
+  };
+  position: Position;
+  width?: CSSProperties['width'];
+} & ObjectType;
+
+export const ProductTour: React.FC<Props> = ({
+  title,
+  description,
+  actions,
+  position,
+  children,
+  img,
+  video,
+  object,
+  width,
+}) => {
+  const media = React.useMemo(() => {
+    if (img) {
+      return <img src={img} />;
+    }
+    if (video) {
+      return <video src={video} />;
+    }
+    return object;
+  }, [img, video, object]);
+
+  const actionsJsx = React.useMemo(() => {
+    return (
+      <ActionsContainer flex={{gap: 6, direction: 'row'}} margin={{top: 8}}>
+        {actions?.next ? <Button onClick={actions.next}>Next</Button> : null}
+        {actions?.dismiss ? <Button onClick={actions.dismiss}>Dismiss</Button> : null}
+      </ActionsContainer>
+    );
+  }, [actions?.next, actions?.dismiss]);
+
+  return (
+    <Popover
+      isOpen={true}
+      placement={position as Placement}
+      modifiers={{
+        arrow: {enabled: true},
+        preventOverflow: {enabled: true},
+      }}
+      minimal={false}
+      content={
+        <>
+          <div />
```

Comment:
Hm why the extra `<div />`s?

Reply:
There was some css on :first-child and :last-child I was trying to avoid